src/Idris/Core/ProofState.hs

```diff
@@ -834,20 +834,20 @@ patbind n ctxt env _ = fail "Can't pattern bind here"
 
 compute :: RunTactic
 compute ctxt env (Bind x (Hole ty) sc) =
-    do return $ Bind x (Hole (normalise ctxt env ty)) sc
+    return $ Bind x (Hole (normalise ctxt env ty)) sc
 compute ctxt env t = return t
 
 hnf_compute :: RunTactic
-hnf_compute ctxt env (Bind x (Hole ty) sc) =
-    do let ty' = hnf ctxt env ty in
+hnf_compute ctxt env (Bind x (Hole ty) sc) = do
```

Comment:
This just seems to be moving the "do" to the dangling position, which we haven't historically had in Idris, rather than eliminating an unnecessary one.